### PR TITLE
feat: implement failed health check logs widget

### PR DESCRIPTION
**Pull request by JedrzejJanasiak**
## Issue

https://gravitee.atlassian.net/browse/APIM-7355

## Description

Implementation of the  Failed Health Check logs widget. 

## Additional context
image

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-arytdmfdgk.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@use 'sass:map';
+@use '@angular/material' as mat;
+@use '@gravitee/ui-particles-angular' as gio;
+
+.card {
+  height: 100%;
+
+  &__title {
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.scss]: is this class in use ?

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+@use 'sass:map';
+@use '@angular/material' as mat;
+@use '@gravitee/ui-particles-angular' as gio;
+
+.card {
+  height: 100%;
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.scss]: do we have to set this value ?

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    public readonly snackBarService: SnackBarService,
+    public readonly destroyRef: DestroyRef,
+  ) {}
+
+  ngOnInit() {
+    this.logs$ = this.apiHealthV2Service.activeFilter().pipe(
+      combineLatestWith(this.filters$),
+      distinctUntilChanged(isEqual),
+      switchMap(([timeRange, tableFilter]) =>
+        this.apiHealthV2Service.getApiHealthCheckLogs(this.apiId, {
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.ts:84]: the loader on refresh / filter change do not gets the isLoading = true ; so the is no loading indicator in UI

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          from: timeRange.from,
+          to: timeRange.to,
+          page: tableFilter.pagination.index,
+          perPage: tableFilter.pagination.size,
+          success: this.showSuccessLogs,
+        }),
+      ),
+      tap(() => {
+        this.isLoading = false;
+      }),
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.ts:98]: we dont handle request failure wiht a toast ;-(

**Review by pes-spyro-soft-com — COMMENTED** [review]
some small items to add

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.scss (hunk 1 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.scss]: good catch, removed

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.scss (hunk 2 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.scss]: not really, removed

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.ts (hunk 4 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/failed-health-checks/failed-health-checks.component.ts:98]: added :)

**Review by pes-spyro-soft-com — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts [hunk 5]
```diff
@@ -117,6 +117,11 @@ export class AvailabilityPerFieldComponent implements OnInit {
 
   mergeGroups = (responseTimes: ApiAverageResponseTime, availability: ApiAvailability): TableData[] => {
     const tableData: TableData[] = [];
+
+    if (responseTimes == null || responseTimes.group == null || availability == null || availability.group == null) {
```
Comment (michel-barret) [inline gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts]: 😉 3 '='
can be write
```suggestion
    if (!responseTimes?.group || !availability?.group) {
```

**Review by michel-barret — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts (hunk 5 above)
Comment (pes-spyro-soft-com) [reply gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts]: but we reducing readability

**Pull request merged**